libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -152,12 +162,12 @@ func NewPodEventer(uuid uuid.UUID, cfg *conf.C, client k8s.Interface, publish fu
 
 	watcher.AddEventHandler(p)
 
-	if nodeWatcher != nil && (config.Hints.Enabled() || metaConf.Node.Enabled()) {
+	if nodeWatcher != nil && config.Hints.Enabled() {
 		updater := kubernetes.NewNodePodUpdater(p.unlockedUpdate, watcher.Store(), &p.crossUpdate)
 		nodeWatcher.AddEventHandler(updater)
 	}
 
-	if namespaceWatcher != nil && (config.Hints.Enabled() || metaConf.Namespace.Enabled()) {
+	if namespaceWatcher != nil && config.Hints.Enabled() {
```

Comment:
Now that I see this line again I realize the `metaConf.Namespace/Node` was not checked in order to initialize the watchers.

Let's take for example the `namespaceWatcher`:
We want to trigger the `updater` even if `metaConf.Namespace.Enabled()!=nil` in case of having Hints enabled. Some details can be found at https://github.com/elastic/beats/pull/25117.

So I believe we need to rethink of this patch more carefully. I see 2 options here:
1) We initialize the watchers if `config.Hints.Enabled() || metaConf.Namespace.Enabled()` anyways in order to have the hints to work as expected.
2) Introduce new settings to disable the Watchers only for users that actually don't have permissions to `watch` on `Namespaces` and/or `Nodes.

The point is that we cannot rely Hint's based autodiscovery functionality with the `add_resource_metadata` setting. That's we had this `||` in this if statement here.

Reply:
> We initialize the watchers if config.Hints.Enabled() || metaConf.Namespace.Enabled() anyways in order to have the hints to work as expected.

I like this option, because this way we can still stop the watchers with `hints.enabled: false`. The main problem as it is now is that we have no way to stop them.
We would have to update the documentation for [Autodiscover](https://www.elastic.co/guide/en/beats/metricbeat/current/configuration-autodiscover.html) with this, as we never mention the option `hints` (only at [this](https://www.elastic.co/guide/en/beats/metricbeat/current/configuration-autodiscover-hints.html) page). 

> Introduce new settings to disable the Watchers only for users that actually don't have permissions to watch on Namespaces and/or Nodes.

Maybe the user would have way too many options when we can already use the ones available.

> A real use case would be the following:

Thanks, it is clear :+1:

Reply:
> We initialize the watchers if config.Hints.Enabled() || metaConf.Namespace.Enabled() anyways in order to have the hints to work as expected.

I commited new changes so now it works like this.

Reply:
Also somehow related: https://github.com/elastic/beats/issues/34717